multiqc/modules/mapdamage/mapdamage.py

```diff
@@ -0,0 +1,291 @@
+""" MultiQC module to parse base misincorporation output from mapdamage2 """
+
+
+import logging
+import os
+from collections import OrderedDict
+
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.plots import linegraph
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """mapDamage module"""
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="mapDamage",
+            anchor="mapdamage",
+            href="https://github.com/ginolhac/mapDamage",
+            info="mapDamage: tracking and quantifying damage patterns in ancient DNA sequences.",
+            doi="https://doi.org/10.1093/bioinformatics/btt193",
+        )
+
+        # Init empty dictionaries
+        self.threepGtoAfreq_data = dict()
+        self.fivepCtoTfreq_data = dict()
+        self.lgdist_fw_data = dict()
+        self.lgdist_rv_data = dict()
+
+        # Find and load log files
+        for f in self.find_log_files("mapdamage", filehandles=True):
+            self.parse_logs(f)
+
+        # Filter to strip out ignored sample names
+        self.threepGtoAfreq_data = self.ignore_samples(self.threepGtoAfreq_data)
+        self.fivepCtoTfreq_data = self.ignore_samples(self.fivepCtoTfreq_data)
+        self.lgdist_fw_data = self.ignore_samples(self.lgdist_fw_data)
+        self.lgdist_rv_data = self.ignore_samples(self.lgdist_rv_data)
+
+        ## Stop computations if there is no data after ignoring samples
+        if (
+            len(self.threepGtoAfreq_data) == 0
+            or len(self.fivepCtoTfreq_data) == 0
+            or len(self.lgdist_fw_data) == 0
+            or len(self.lgdist_rv_data) == 0
+        ):
+            raise UserWarning
```

Comment:
@TCLamnidis - is this meant to be `or`? Currently this will not produce a report if any of these four are empty. 

Typically I'd be used to the opposite - make a report if we found any data at all..

Probably doesn't make any difference, but I thought I'd ask.

Reply:
```suggestion
        ## Stop computations if there is no data after ignoring samples
        if (
            len(self.threepGtoAfreq_data) == 0
            and len(self.fivepCtoTfreq_data) == 0
            and len(self.lgdist_fw_data) == 0
            and len(self.lgdist_rv_data) == 0
        ):
            raise UserWarning
```